Wip/edc 2691

DanielM235 opened the pull request.

Commit:
[EDC-2695] Add project bundler and tools

- ts-config
- webpack
- eslint
- jest
- travis file

Commit:
[EDC-2696][EDC-2701] Popover initialisation

[EDC-2696] - Configuration loading
[EDC-2697] - Initialize main inputs (mainKey, subKey)
[EDC-2698] - Common options
[EDC-2699] - Fail behavior
[EDC-2700] - Style customization
[EDC-2701] - Language handling

Unit Tests
set version to 2.0.0-beta-0

Story [EDC-2691]

Commit:
[EDC-2691] Add storybook v6

- include a mini doc
- downgrade webpack to v4 due to incompatibility with storybook (https://github.com/storybookjs/storybook/issues/9216)

greg7859 commented on .storybook/tsconfig.json:
```diff
@@ -0,0 +1,84 @@
+    // "declaration": true,                   /* Generates corresponding '.d.ts' file. */
+    // "declarationMap": true,                /* Generates a sourcemap for each corresponding '.d.ts' file. */
+    "declaration": false,
+    // "sourceMap": true,                     /* Generates corresponding '.map' file. */
```
Remove this line.
Why do you keep the commented line?

DanielM235 replied:
for easier memo access, but it does make the configuration less readable, will remove them

DanielM235 replied:
done

greg7859 commented on package.json:
```diff
@@ -1,12 +1,68 @@
+    "less-plugin-clean-css": "^1.5.1",
+    "mini-css-extract-plugin": "^1.0.0",
+    "puppeteer": "^2.1.1",
+    "react": "^17.0.0",
```
Do you need react for the JS library ?

DanielM235 replied:
it's for storybook, some of their extensions require it

greg7859 commented on tsconfig.json:
```diff
@@ -0,0 +1,86 @@
+      "es2015"
+    ],
+    /* Specify library files to be included in the compilation. */
+    // "allowJs": true,                       /* Allow javascript files to be compiled. */
```
Do you need to keep the commented line (unused configuration)?

DanielM235 replied:
done

Commit:
[EDC-2703] Update README.md, v2.0.0

- edc-popover-utils v1.0.0
- remove unnecessary comments
- set version to 2.0.0
